platform/ios/MGLMapView.mm

```diff
@@ -138,69 +137,73 @@ - (instancetype)initWithFrame:(CGRect)frame accessToken:(NSString *)accessToken
 
     if (self && [self commonInit])
     {
-        if (accessToken) [self setAccessToken:accessToken];
-        if (styleURL) [self setStyleURL:styleURL];
+        self.accessToken = accessToken;
+        self.styleURL = styleURL;
     }
 
     return self;
 }
 
-- (instancetype)initWithFrame:(CGRect)frame accessToken:(NSString *)accessToken
-{
-    return [self initWithFrame:frame accessToken:accessToken styleJSON:nil];
-}
-
 - (instancetype)initWithCoder:(NSCoder *)decoder
 {
     self = [super initWithCoder:decoder];
 
     if (self && [self commonInit])
     {
+        self.styleURL = nil;
         return self;
     }
 
     return nil;
 }
 
+- (NSString *)accessToken
+{
+    return @(mbglMap->getAccessToken().c_str()).mgl_stringOrNilIfEmpty;
+}
+
 - (void)setAccessToken:(NSString *)accessToken
 {
-    if (accessToken)
-    {
-        mbglMap->setAccessToken((std::string)[accessToken UTF8String]);
-        [MGLMapboxEvents setToken:accessToken];
-    }
+    mbglMap->setAccessToken((std::string)[accessToken UTF8String]);
+    [MGLMapboxEvents setToken:accessToken.mgl_stringOrNilIfEmpty];
 }
 
-- (void)setStyleJSON:(NSString *)styleJSON
++ (NSSet *)keyPathsForValuesAffectingStyleURL
 {
-    if ( ! styleJSON)
-    {
-        [self useBundledStyleNamed:[[[MGLDefaultStyleName lowercaseString]
-                                        stringByAppendingString:@"-"]
-                                        stringByAppendingString:MGLStyleVersion]];
-    }
-    else
-    {
-        mbglMap->setStyleJSON((std::string)[styleJSON UTF8String]);
-    }
+    return [NSSet setWithObjects:@"mapID", @"accessToken", nil];
+}
+
+- (NSURL *)styleURL
+{
+    NSString *styleURLString = @(mbglMap->getStyleURL().c_str()).mgl_stringOrNilIfEmpty;
+    return [NSURL URLWithString:styleURLString];
 }
 
 - (void)setStyleURL:(NSURL *)styleURL
 {
+    if (_isTargetingInterfaceBuilder) return;
+    
+    if ( ! styleURL)
+    {
+        styleURL = MGLURLForBundledStyleNamed([NSString stringWithFormat:@"%@-v%@",
+                                               MGLDefaultStyleName.lowercaseString,
+                                               MGLStyleVersion]);
+    }
+    
     std::string styleURLString([[styleURL absoluteString] UTF8String]);
 
     if ( ! [styleURL scheme])
     {
-        mbglMap->setStyleURL(std::string("asset://") + styleURLString);
-    }
-    else
-    {
-        mbglMap->setStyleURL(styleURLString);
+        styleURLString = std::string("asset://") + styleURLString;
```

Comment:
I think we should get rid of this case. We can easily be consistent internally about using the asset: scheme. If the developer then passes in a relative URL, we can treat it as being relative to the main bundle rather than MapboxGL.bundle.